nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/AbstractMongoProcessor.java

```diff
@@ -102,8 +107,26 @@
             .defaultValue(WRITE_CONCERN_ACKNOWLEDGED)
             .build();
 
+    static final PropertyDescriptor RESULTS_PER_FLOWFILE = new PropertyDescriptor.Builder()
+            .name("results-per-flowfile")
+            .displayName("Results Per FlowFile")
+            .description("How many results to put into a flowfile at once. The whole body will be treated as a JSON array of results.")
+            .required(false)
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .defaultValue("1")
+            .build();
+
+    static final PropertyDescriptor BATCH_SIZE = new PropertyDescriptor.Builder()
+            .name("Batch Size")
```

Comment:
Generally it's preferred to add a displayname and make name without whitespaces, like "batch-size"

Reply:
I know, but I didn't write the class and it's been around for a while. I wouldn't want to risk breaking others' flows over that.

Reply:
In this case, the guidance for a new `PropertyDescriptor` in an existing class (or new class) is to add a `displayName` value and provide a "machine-readable" (i.e. no space) value for `name`. When modifying an *existing* `PropertyDescriptor`, adding a `displayName` value is encouraged, but modifying the `name` will break backward compatibility with existing flows.

Reply:
Ok. Made that change and added the same change to a few other property descriptors while I was doing it.